ocaml/typing/mode.ml

```diff
@@ -1533,8 +1491,8 @@ type ('mo, 'como) monadic_comonadic =
     comonadic : 'como
   }
 
-module Value = struct
-  module Comonadic = Comonadic_with_regionality
+module Mode (Areality : Areality) = struct
```

Comment:
I don't have a strong opinion, but it might be worthwhile to think of a more descriptive name than `Mode`. Maybe `Value_with`? Similar to `Comonadic_with`.

The `Alloc` mode is a bad name anyway, since it's more about "a dumbed down version of `Value` exposed to the user" than about "allocation".

Reply:
Good suggestion.

I'll leave renaming `Alloc` for another PR, but I'd support it.